packages/mdc-menu/foundation.ts

```diff
@@ -131,6 +132,27 @@ export class MDCMenuFoundation extends MDCFoundation<MDCMenuAdapter> {
     this.defaultFocusState_ = focusState;
   }
 
+  /**
+   * Selects the list item at `index` within the menu.
+   * @param index Index of list item within the menu.
+   */
+  setSelectedIndex(index: number) {
+    if (!this.isIndexInRange_(index)) {
+      throw new Error('MDCMenuFoundation: No list item at specified index.');
+    }
+    const listItem = this.adapter_.getListItemAtIndex(index);
```

Comment:
I understand that this adapter API was introduced to be consistent with other adapter APIs in this foundation. IMO foundation should avoid accessing DOM directly and this needs to be moved over to adapter. For example, `getSelectionGroup_` logic should be moved to adapter.

I created a new branch ([feat/menu/selected-item-selection-group_cleanup_techdebt](https://github.com/material-components/material-components-web/compare/feat/menu/selected-item-selection-group_cleanup_techdebt?expand=1)) based off of your branch to clean up the tech debt that we had earlier with menu foundation. PTAL, Feel free to merge this into your branch and make required changes if it looks good.

Thanks!

Reply:
It sounds like I don't need to merge your branch into mine. I can just work off of your branch, since you branched off of this one. Is that correct? I think your changes make sense. And yes we shouldn't be referencing the DOM in the foundation.

Reply:
You can use the same PR if you merge my branch into yours.